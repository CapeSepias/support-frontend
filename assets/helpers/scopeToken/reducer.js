// @flow

// ----- Types ----- //

export type ScopeToken = ?string;


// ----- Setup ----- //

const initialState: ScopeToken =
  window.guardian && window.guardian.scopeToken ? window.guardian.scopeToken : null;


// ----- Reducer ----- //

export default function scopeTokenReducer(state: ScopeToken = initialState): ScopeToken {
  return state;
}
