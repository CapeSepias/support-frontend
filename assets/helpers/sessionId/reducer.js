// @flow
import uuidv4 from 'uuid';
import * as storage from 'helpers/storage';

// ----- Types ----- //

export type SessionId = ?string;


// ----- Setup ----- //

function getAndStoreId(keyName) {
  let value = storage.getSession(keyName);
  if (value === null) {
    value = uuidv4();
    storage.setSession(keyName, value);
  }
  return value;
}

const initialState: SessionId = getAndStoreId('sessionId');

// ----- Reducer ----- //

export default function sessionIdReducer(state: SessionId = initialState): SessionId {
  return state;
}
