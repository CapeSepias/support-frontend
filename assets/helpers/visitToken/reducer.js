// @flow

// ----- Imports ----- //

import type { Action } from './actions';


// ----- Types ----- //

export type VisitToken = {
    token: ?string,
};


// ----- Setup ----- //

const initialState: VisitToken = window.guardian && window.guardian.visitToken ? {
  token: window.guardian.visitToken,
} : {
  token: null,
};


// ----- Reducer ----- //

export default function visitTokenReducer(
  state: VisitToken = initialState,
  action: Action,
): VisitToken {

  switch (action.type) {

    case 'SET_VISIT_TOKEN':
      return Object.assign({}, state, { token: action.name });

    default:
      return state;

  }

}
