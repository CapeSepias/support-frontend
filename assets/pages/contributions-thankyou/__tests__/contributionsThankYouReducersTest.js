// @flow
import reducer from '../contributionsThankYouReducer';


describe('Thank you Reducer', () => {

  it('should return the initial state', () => {
    console.log(reducer(undefined, {}));
    expect(reducer(undefined, {})).toMatchSnapshot();
  });

  it('should handle CONSENT_API_ERROR', () => {

    const consentApiError = true;
    const action = {
      type: 'CONSENT_API_ERROR',
      consentApiError,
    };

    const newState = reducer(undefined, action);

    expect(newState.thankYouState.consentApiError).toEqual(consentApiError);
  });
});
