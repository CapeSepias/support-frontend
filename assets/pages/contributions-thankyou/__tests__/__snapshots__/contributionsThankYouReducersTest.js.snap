// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Thank you Reducer should return the initial state 1`] = `
Object {
  "csrf": Object {
    "token": null,
  },
  "thankYouState": Object {
    "consentApiError": false,
  },
  "user": Object {
    "displayName": "",
    "email": "",
    "firstName": "",
    "gnmMarketing": false,
    "id": "",
    "isPostDeploymentTestUser": false,
    "isTestUser": null,
    "lastName": "",
  },
}
`;
